### ISSUE TITLE
Static payload fails to load after navigating back to the page we hydrated from with non-ASCII path and trailing slash

### ISSUE THREAD

**Issue by awaken1ng** [nuxt/nuxt#25508]
### Environment

- Operating System: Linux
- Node Version:     v21.6.1
- Nuxt Version:     3.9.3
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@10.4.0
- Builder:          -
- User Config:      ssr
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-5bmuhv?startScript=generate,preview

- Generate static build
- Start the preview server
- Click on `to wide foo (with trailing slash)`
- Reload the (internal browser) page
- Click on `to index`
- Click on `to wide foo (with trailing slash)` or `to wide foo (without trailing slash)`

### Describe the bug

Basically the title, after initial loading on a page with non-ASCII path and trailing slash (e.g. `/dynamic/ｆｏｏ/`),
then navigating somewhere (e.g. `/`), and coming back to the initial page (`/dynamic/ｆｏｏ/` or `/dynamic/ｆｏｏ`)
results in static payload failing to load and falling back to API call (which fails, being a static build)

### Additional context

3.8.0 works fine, 3.8.1 is the first version that exhibits this issue, bisected to 62b7917d2ca21794a54e69b9667b1ac8547b2ed3.

**Comment by danielroe** [comment]
This was resolved in https://github.com/nuxt/nuxt/pull/24516.
